plugins/chat/app/services/trash_channel.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+module Chat
+  module Service
+    # Service responsible for trashing a chat channel.
+    # Note the slug is modified to prevent collisions.
+    #
+    # @example
+    #  Chat::Service::TrashChannel.call(channel: channel, guardian: guardian)
+    #
+    class TrashChannel
+      include Base
+
+      # @!method call(channel:, guardian:)
+      #   @param [ChatChannel] channel
+      #   @param [Guardian] guardian
+      #   @return [Chat::Service::Base::Context]
+
+      DELETE_CHANNEL_LOG_KEY = "chat_channel_delete"
+
+      before_contract { guardian(:can_delete_chat_channel?) }
+
+      contract do
+        attribute :channel
+        validates :channel, presence: true
```

Comment:
I noticed we are missing the `name_confirmation` here and in JS, not sure why this is removed? 

Also I was thinking of validation. We can validate the presence of it here, but how do we raise a validation error if the `name_confirmation` does not match the channel name? Do we just do an AM validation like so? (I am pretty sure we do):

```ruby
validate :name_confirmation_matches

def name_confirmation_matches
  errors.add(:name_confirmation, :invalid) if channel.title(guardian.user).downcase != name_confirmation
end

Reply:
I don't think it needs server confirmation that's all :p Frontend only check is fine by me, it's footgun territory if you start to hack  your own js to be able to send a wrong name confirmation to the server.

But yes for the record you could do AM validation here, although guardian is not accessible here, I will wait to see if we actually need this at some point, because here technically you need a user not a guardian for example.

Reply:
Fair enough, I guess you are right that it's a frontend only thing. I think I originally made it server-side as well because GitHub sends the confirmation in the delete repo payload. Not a hill I will die on though :)